### PR TITLE
Added documentation for dynamic types workaround

### PR DESCRIPTION
**Pull request by alexthe2**
*Please list issues fixed by this PR here, using format "Fixes #the-issue-number".*

## Checklist

- [x] An issue to be fixed by this PR is listed above. https://github.com/fzyzcjy/flutter_rust_bridge/issues/798
- [ ] CI is passing.

### Not applicable
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [x] If this PR adds/changes features, documentations (in the `./book` folder) are updated.

## Remark for PR creator

* New contributors will be blocked by GitHub from running CI, but you can use [a trick](https://github.com/fzyzcjy/flutter_rust_bridge/pull/663#discussion_r962150638) to workaround this, and verify your code using CI by yourself.
* If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

File: book/src/feature/lang_dynamic.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+other fields:
+
+```rust,noplayground
+struct MyStruct {
+    a: Optional<u32>,
+    b: Optional<String>,
+}
+
+struct DataStruct {
+    msg:  String
```
Comment (fzyzcjy) [inline book/src/feature/lang_dynamic.md:20]: uber-nit: missing ","

File: book/src/feature/lang_dynamic.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    U32(u32),
+    String(String),
+}
+```
+
+And then you can define a struct that holds this `enum`:
+
+```rust,noplayground
+struct MyStruct {
+    msg:  String
```
Comment (fzyzcjy) [inline book/src/feature/lang_dynamic.md:38]: uber-nit: missing ","

File: book/src/feature/lang_dynamic.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+## Example
+
+Let's say you have a struct that can hold either a `u32` or a `String` and some
+other fields:
+
+```rust,noplayground
+struct MyStruct {
+    a: Optional<u32>,
+    b: Optional<String>,
+}
```
Comment (fzyzcjy) [inline book/src/feature/lang_dynamic.md:17]: well, this does not look like a reasonable usage

Comment (alexthe2) [reply book/src/feature/lang_dynamic.md:17]: I was thinking about that as well, but isn't this the alternate (worse) design?

Comment (fzyzcjy) [reply book/src/feature/lang_dynamic.md:17]: Ok, then just say this is a worse design ;)

**Comment by fzyzcjy** [comment]
@all-contributors please add @alexthe2 for doc

**Pull request merged**